src/galileo/__future__/metric.py

```diff
@@ -657,53 +447,397 @@ def to_legacy_metric(self) -> LegacyMetric:
         """
         return LegacyMetric(name=self.name, version=self.version)
 
-    def to_local_metric_config(self) -> LocalMetricConfig:
+    def __str__(self) -> str:
+        """String representation of the metric."""
+        type_name = self.__class__.__name__
+        scorer_type_str = self.scorer_type.value if self.scorer_type else "unknown"
+        return f"{type_name}(name='{self.name}', id='{self.id}', scorer_type='{scorer_type_str}')"
+
+    def __repr__(self) -> str:
+        """Detailed string representation of the metric."""
+        type_name = self.__class__.__name__
+        return f"{type_name}(name='{self.name}', id='{self.id}')"
+
+
+# ============================================================================
+# Concrete Metric Types
+# ============================================================================
+
+
+class LlmMetric(Metric):
+    """
+    LLM-based metric with custom prompt templates.
+
+    This metric type allows you to create custom metrics evaluated by an LLM
+    judge using a prompt template.
+
+    Attributes
+    ----------
+        prompt (str | None): Prompt template for the LLM scorer.
+        model (str | None): Model name to use for scoring.
```

Comment:
should we have something here that indicates that the model must align with a model name avaialble in Galileo?  Will we have an `Integration` or `Model` top level class in the python SDK that we could reference here instead of just a string?